hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3324,8 +3324,16 @@ TermIndex installCheckpoint(String leaderId, Path checkpointLocation,
       omRatisServer.getOmStateMachine().unpause(lastAppliedIndex, term);
       LOG.info("Reloaded OM state with Term: {} and Index: {}", term,
           lastAppliedIndex);
-    } catch (IOException ex) {
+    } catch (Exception ex) {
       String errorMsg = "Failed to reload OM state and instantiate services.";
+      // Delete the backup DB if exists and then terminate OM.
+      try {
+        if (dbBackup != null) {
+          FileUtils.deleteFully(dbBackup);
+        }
+      } catch (Exception e) {
+        LOG.error("Failed to delete the backup of the original DB {}", dbBackup);
+      }
```

Comment:
Bharat, in this scenario where the OM state was not reloaded with new checkpoint, it would be better to keep the backup also in place. Just in case the OM needs to be manually reversed back to old state.

Reply:
Ya thought of it, but my thought process here is when OM restarts it will get a new install Snapshot and try again. Because when OM it starts, anyway it cannot use this backup path, until some one manual replacing.

But I think to be on safer side, will remove this code for now.

Reply:
Done.